server/xpub-controller/helpers/files.js

```diff
@@ -0,0 +1,161 @@
+const logger = require('@pubsweet/logger')
+const FileModel = require('@elifesciences/xpub-model').File
+
+class FilesHelper {
+  constructor(config, scienceBeamApi) {
+    this.config = config
+    this.scienceBeamApi = scienceBeamApi
+  }
+
+  validateFileSize(fileSize) {
+    if (fileSize > this.config.get('fileUpload.maxSizeMB') * 1e6) {
+      throw new Error(
+        `File size shouldn't exceed ${this.config.get(
+          'fileUpload.maxSizeMB',
+        )}MB`,
+      )
+    }
+  }
+
+  static async generateFileEntity(file, manuscriptId) {
+    const { stream, filename, mimetype: mimeType } = await file
+
+    const fileEntity = await new FileModel({
+      manuscriptId,
+      url: `manuscripts/${manuscriptId}`,
+      filename,
+      type: 'MANUSCRIPT_SOURCE_PENDING',
+      mimeType,
+    }).save()
+    return {
+      stream,
+      fileEntity,
+    }
+  }
+
+  static publishPredictedProgress(
+    pubsub,
+    ON_UPLOAD_PROGRESS,
+    startedTime,
+    predictedTime,
+    manuscriptId,
+  ) {
+    if (
+      typeof pubsub.publish === 'function' &&
```

Comment:
This `typeof` check looks like typing to me. So far we've avoided type checking and I think we should continue avoiding it at this point. The rest of the validation looks 👌